main/views.py

```diff
@@ -75,71 +145,339 @@ def add_service_activity():
+            service_id=service.id)
+        db.session.add(spec_service)
+        return jsonify({
+            'status': 'ok',
+            'service': {
+                'name': service.title,
+                'id': service.id
+            }
+        })
+
+    return jsonify({
+        'status': 'error',
+        'errors': form.errors
+    })
+
+
+@app.route('/add_specialist_service', methods=['POST'])
+@login_required
+def add_searched_services():
+    services = []
+    for service_id in json.loads(request.data).get('selected_service_ids', []):
+        service = Service.query.filter_by(id=service_id).first()
+        if not service:
+            return jsonify({
+                'status': 'error',
+            })
+
+        if service not in current_user.specialist.services.all():
+            services.append({
+                'name': service.title,
+                'id': service_id
+            })
+
+            spec_service = SpecialistService(
+                specialist_id=current_user.specialist.id,
+                service_id=service_id)
+            db.session.add(spec_service)
+
+    return jsonify({
+        'status': 'ok',
+        'services': services
+    })
+
+
+@app.route('/create_specialist', methods=['POST'])
+@login_required
+def create_specialist():
+    form = SpecialistForm()
+    if form.validate():
+        current_user.phone_number = form.phone.data
+        specialist = Specialist(
+            user=current_user,
+            description=form.description.data,
+            experience=form.experience.data)
+
+        db.session.add(specialist)
+
+        return jsonify({
+            'status': 'ok'
+        })
+    else:
+        return jsonify({
+            'status': 'error',
+            'errors': form.errors
+        })
+
+
+class LoginView(FormView):
+    form_class = LoginForm
+    template_name = 'Login.html'
+    
+    def get(self, *args, **kwargs):
+        if current_user.is_authenticated:
+            return redirect(url_for('user_profile', user_id=current_user.id))
+        return super(LoginView, self).get(*args, **kwargs)
+
+    def form_invalid(self, form):
+        return jsonify({
+            'status': 'error',
+            'errors': form.errors
+        })
+
+    def form_valid(self, form):
+        try:
+            user = User.query.filter(
+                User.email == form.email.data).one()
+        except NoResultFound:
+            return jsonify({
+                'status': 'error',
+                'errors': {
+                    'password':
+                        'User with entered email and password does not exist'
                 }
-            ])
-        customers = get_model_column_values(
-            Customer,
-            columns=[
-                {
-                    'dict_key': 'id', 'column': 'id'
-                },
-                {
-                    'dict_key': 'name', 'column': 'name'
+            })
+
+        if user.password != form.password.data:
```

Comment:
I think you mentioned it so I'm leaving the comment as a reminder: saving raw passwords isn't safe